src/garage/torch/modules/multi_headed_mlp_module.py

```diff
@@ -0,0 +1,125 @@
+import torch.nn as nn
+
+
+class MultiHeadedMLPModule(nn.Module):
+    ERROR_MESSAGE = '{} should be either an integer or a collection of length n_heads ({}), but {} provided.'
+
+    """
+    MultiHeadedMLPModule Model.
+    
+
+    Args:
+        n_heads (int) : Number of different output layers 
+        input_dim (int) : Dimension of the network input.
+        output_dims (int, list, or tuple): Dimension of the network output.
+        hidden_sizes (list[int]): Output dimension of dense layer(s).
+            For example, (32, 32) means this MLP consists of two
+            hidden layers, each with 32 hidden units.
+        hidden_nonlinearity (callable): Activation function for intermediate
+            dense layer(s). It should return a torch.Tensor. Set it to
+            None to maintain a linear activation.
+        hidden_w_init (callable): Initializer function for the weight
+            of intermediate dense layer(s). The function should return a
+            torch.Tensor.
+        hidden_b_init (callable): Initializer function for the bias
+            of intermediate dense layer(s). The function should return a
+            torch.Tensor.
+        output_nonlinearity (callable, list, or tuple): Activation function for output dense
+            layer. It should return a torch.Tensor. Set it to None to
+            maintain a linear activation.
+            Size of the parameter should be 1 or equal to n_head
+        output_w_init (callable, list, or tuple): Initializer function for the weight
+            of output dense layer(s). The function should return a
+            torch.Tensor.
+            Size of the parameter should be 1 or equal to n_head
+        output_b_init (callable, list, or tuple): Initializer function for the bias
+            of output dense layer(s). The function should return a
+            torch.Tensor.
+            Size of the parameter should be 1 or equal to n_head
+        layer_normalization (bool): Bool for using layer normalization or not.
+
+    Return:
+        The list of output torch. Tensor of the MLP
+    """
+    def __init__(self,
+                 n_heads,
+                 input_dim,
+                 output_dims,
+                 hidden_sizes,
+                 hidden_nonlinearity=nn.ReLU,
+                 hidden_w_init=nn.init.xavier_normal_,
+                 hidden_b_init=nn.init.zeros_,
+                 output_nonlinearities=None,
+                 output_w_inits=nn.init.xavier_normal_,
+                 output_b_inits=nn.init.zeros_,
+                 layer_normalization=False):
+        super().__init__()
+
+        self._layers = nn.ModuleList()
+
+        if n_heads < 1:
+            raise ValueError("n_head should be greater than or equal to 1 but {} provided.".format(n_heads))
+
+        if isinstance(output_dims, (list, tuple)):
```

Comment:
can you put this repeated incantation into a helper function? it would help you share the error message, too

Reply:
fixed